src/PhpUnit/UriInterfaceTestsTrait.php

```diff
@@ -0,0 +1,290 @@
+<?php
+
+namespace Maks3w\Psr7Assertions\PhpUnit;
+
+use PHPUnit_Framework_Assert as Assert;
+use Psr\Http\Message\UriInterface;
+
+/**
+ * Provide PHPUnit test methods for Psr\Http\Message\UriInterface constraints.
+ */
+trait UriInterfaceTestsTrait
+{
+    /**
+     * @return Psr\Http\Message\UriInterface
+     */
+    abstract public function createUriObject($url);
+
+    public function testUriImplementsInterface()
+    {
+        Assert::assertInstanceOf('Psr\Http\Message\UriInterface', $this->createUriObject(''));
+    }
+
+    /**
+     * @dataProvider uriProvider
+     * @group scheme
```

Comment:
Sorry the `@group` is a "left-over" from my test suite because it helped me test the methods and also compare methods implementations in different packages. I will remove them Once all you feedback are done. 
I also think the tests for the mutators methods ``with*` can be improved also. I'm working on them to. But I needed you feedback first on this PR